loadgen/generic_executor.go

```diff
@@ -106,24 +127,27 @@ func (g *genericRun) Run(ctx context.Context) error {
 		go func() {
 			startTime := time.Now()
 			err := g.executor.Execute(ctx, run)
-			// Only log/wrap/send to channel if context is not done
-			if ctx.Err() == nil {
-				if err != nil {
-					err = fmt.Errorf("iteration %v failed: %w", run.Iteration, err)
-					g.logger.Error(err)
-				}
-				select {
-				case <-ctx.Done():
-				case doneCh <- err:
-					// Record/log here, not if it was cut short by context complete
-					g.executeTimer.Record(time.Since(startTime))
-				}
+			if err != nil {
+				err = fmt.Errorf("iteration %v failed: %w", run.Iteration, err)
+				g.logger.Error(err)
+			}
+			select {
+			case <-ctx.Done():
```

Comment:
Should check durationCtx instead or this can deadlock.

Reply:
I don't think that's right - this needs to be like this. If we exited when durationCtx was done, always, then we would fail to properly wait on outstanding workflows at the end. I don't see how it could deadlock - it could take forever if the runs never complete (and no timeout is specified), but that's kinda the point.